lib/charms/operator_libs_linux/v0/apt.py

```diff
@@ -1107,12 +1118,14 @@ def disable(self) -> None:
                 " Please raise an issue if you require this feature."
             )
         searcher = f"{self.repotype} {self.make_options_string()}{self.uri} {self.release}"
-        with fileinput.input(self._filename, inplace=True) as lines:
-            for line in lines:
-                if re.match(rf"^{re.escape(searcher)}\s", line):
-                    print(f"# {line}", end="")
-                else:
-                    print(line, end="")
+        with tracer.start_as_current_span("disable source") as span:
```

Comment:
What's the motivation for the span here? Is it just because it's interacting with external files? Or because you suspect this might take a significant amount of time? Something else?

Reply:
My original thinking was that it has system-wide effect.

The secondary was that the file could be arbitrarily large and processing could take arbitrarily long. Then again, these files are never massive.